TestModels/CallingAWSSDKFromLocalService/test/SimpleCallingAWSSDKFromLocalServiceImplTest.dfy

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+include "../src/Index.dfy"
+include "../src/WrappedSimpleCallingAWSSDKFromLocalServiceImpl.dfy"
+
+module SimpleCallingAWSSDKFromLocalServiceImplTest {
+  import Com.Amazonaws.Dynamodb
+  import Com.Amazonaws.Kms
+  import SimpleCallingAWSSDKFromLocalService
+
+  // For call to DDB
+  const TABLE_ARN_SUCCESS_CASE := "arn:aws:dynamodb:us-west-2:370957321024:table/TestTable"
+  const TABLE_ARN_FAILURE_CASE := "arn:aws:dynamodb:us-west-2:370957321024:table/TestTableFailure"
+  const NONEXISTENT_TABLE_NAME := "NONEXISTENT_Table"
+
+  // For call to KMS
+  const KEY_ID_SUCCESS_CASE := "arn:aws:kms:us-west-2:658956600833:key/b3537ef1-d8dc-4780-9f5a-55776cbb2f7f"
+  const INVALID_KEY_ID := "arn:aws:kms:us-west-2:658956600833:key/b3537ef1-d8dc-4780-9f5a-invalidkeyid"
+  const NONEXISTENT_KEY_ID := "arn:aws:kms:us-west-2:658956600833:key/b3537ef1-d8dc-4780-9f5a-55776cbb2f7g"
+  // The string "asdf" as bytes
+  const PLAIN_TEXT: Kms.Types.PlaintextType := [ 97, 115, 100, 102 ]
+
+  import opened SimpleCallingawssdkfromlocalserviceTypes
+  import opened Wrappers
+  method{:test} CallDDBScan(){
+    var client :- expect SimpleCallingAWSSDKFromLocalService.SimpleCallingAWSSDKFromLocalService();
+    TestCallDDBScan_Success(client);
+    TestCallDDBScan_Failure(client);
+  }
+
+  method TestCallDDBScan_Success(client: ISimpleCallingAWSSDKFromLocalServiceClient)
+      requires client.ValidState()
+      modifies client.Modifies
+      ensures client.ValidState()
+  {
+    var ddbClient :- expect Dynamodb.DynamoDBClient();
+    var resSuccess := client.CallDDBScan(SimpleCallingAWSSDKFromLocalService.Types.CallDDBScanInput(ddbClient := ddbClient, tableArn := TABLE_ARN_SUCCESS_CASE));
+
+    expect resSuccess.Success?;
+    expect resSuccess.value.itemOutput != -1;
+  }
+
+  method TestCallDDBScan_Failure(client: ISimpleCallingAWSSDKFromLocalServiceClient)
+    requires client.ValidState()
+    modifies client.Modifies
+    ensures client.ValidState()
+  {
+    var ddbClient :- expect Dynamodb.DynamoDBClient();
+    var resFailure := client.CallDDBScan(SimpleCallingAWSSDKFromLocalService.Types.CallDDBScanInput(ddbClient := ddbClient, tableArn := TABLE_ARN_FAILURE_CASE));
+
+    expect resFailure.Failure?;
```

Comment:
We can also check for Opaque error over here but first we need to check if its DynamoDB or KMS error for Dafny verification to pass and for specifically Opaque error its little fragile in Go codegen. So, it doesn't have it but lmk if you think otherwise.

Reply:
Let's add a TODO to verify the error that get's thrown? I guess this is blocked on our codegen error conversion fix?

Reply:
Also, DDB give opaque error when trying to access table not in the account for user with access to limited number of table because they can't say resource not found (because user does not have access to all table) and they give AccessDeniedException which is not DDB exception but an IAM exception.